### PR TITLE
Improved error message when embedding a Youtube video with embedding disabled

### LINKED ISSUE #16048: Improve error message for YouTube videos that disabled embedding
### Issue Summary

EDIT: I've figured out that this specific YouTube doesn't allow embedding. Maybe you could improve the error message shown to the user  if you get the `Unauthorized` message (see below) so this causes less confusion. I have also updated the issue title to reflect this. I'm leaving the following text here as was to leave the context.

---

In general, YouTube embedding works. But this specific returns an error: `https://www.youtube.com/watch?v=5f-JlzBuUUU`.

CleanShot 2022-12-25 at 20 10 24@2x

The request is the following:
```
https://cassandradispatch.org/ghost/api/admin/oembed/?url=https://www.youtube.com/watch?v=5f-JlzBuUUU&type=embed
```

The error response is the following:

```json
{
	"errors": [
		{
			"message": "Internal server error, cannot read oembed.",
			"context": "invalid json response body at https://www.youtube.com/oembed?format=json&url=https%3A%2F%2Fwww.youtube.com%2Fwatch%3Fv%3D5f-JlzBuUUU reason: Unexpected token U in JSON at position 0",
			"type": "InternalServerError",
			"details": null,
			"property": null,
			"help": null,
			"code": null,
			"id": "e01223d0-8487-11ed-94c3-8df7a8a46270",
			"ghostErrorCode": null
		}
	]
}
```

This is becaue the request
`https://www.youtube.com/oembed?format=json&url=https%3A%2F%2Fwww.youtube.com%2Fwatch%3Fv%3D5f-JlzBuUUU`

Returns `Unauthorized` as the response body.

### Steps to Reproduce

1. Create a new post (or edit an existing one)
2. Click the "plus" sign to add a new bloc and choose "YouTube"
3. Paste this URL: `https://www.youtube.com/watch?v=5f-JlzBuUUU`
4. See the error

### Ghost Version

5.26.3

### Node.js Version

16.18.1

### How did you install Ghost?

Hetzner, Ubuntu 22.04.1

### Database type

MySQL 8

### Browser & OS version

_No response_

### Code of Conduct

- [X] I agree to be friendly and polite to people in this repository

### PR DESCRIPTION
**Pull request by cmraible**
fixes TryGhost/Ghost#16048

- When attempting to embed a Youtube video that has had embedding disabled by its owner/author, Ghost displayed a generic error message that didn't indicate the reason for the failed embed.

- This change updated the error message when Youtube (or any provider) returns 401: Unauthorized to indicate that the owner of the resource has explicity disabled embedding.

### CODE REVIEW

File: ghost/admin/lib/koenig-editor/addon/components/koenig-card-embed.js
Comment (kevinansfield) [inline ghost/admin/lib/koenig-editor/addon/components/koenig-card-embed.js:215]: Rather than checking the message text it should be possible to use the `isUnauthorizedError()` function from the [ember-ajax addon](https://github.com/ember-cli/ember-ajax#error-handling) we use, which should be a little more robust as it uses the response status rather than message. Example here https://github.com/TryGhost/Ghost/blob/main/ghost/admin/app/routes/application.js#L92

**Review by kevinansfield — APPROVED** [review]

File: ghost/admin/lib/koenig-editor/addon/components/koenig-card-embed.js
Comment (cmraible) [reply ghost/admin/lib/koenig-editor/addon/components/koenig-card-embed.js]: Awesome thanks for the tip!

**Comment by cmraible** [comment]
Status update: this PR is ready to merge, but pending coordination with core team on testing/deployment strategy.

Last time we tried to upgrade the oembed-parser library, we encountered a bunch of 422 errors and decided to rollback. This time, we want to test on a small subset of Pro users before deploying widely, so holding this PR until we have a solid window to test and rollout.

**Comment by cmraible** [comment]
😅 I let this PR go pretty stale and was having a hard time reconciling conflicts so I just started over with a new branch — closing this PR in favor of https://github.com/TryGhost/Ghost/pull/16374

**Pull request closed without merging**
